packages/react-components/react-breadcrumb/src/utils/partitionBreadcrumbItems.ts

```diff
@@ -1,10 +1,16 @@
 const DEFAULT_OVERFLOW_INDEX = 1;
 const DEFAULT_MAX_DISPLAYED_ITEMS = 6;
 
+type Enumerate<N extends number, Acc extends number[] = []> = Acc['length'] extends N
+  ? Acc[number]
+  : Enumerate<N, [...Acc, Acc['length']]>;
+type NumberRange<F extends number, T extends number> = Exclude<Enumerate<T>, Enumerate<F>>;
+type PusitiveNumbers = NumberRange<0, 100>;
+
```

Comment:
Not sure if it's needed, cause it looks like overengineering. I would like to hear your thoughts

Reply:
if TS doesnt provide build in solution with good DX it shouldn't be used for every API constraint from my experience. this case is rather great candidate for lint rule IMO.

Reply:
removed it :)